packages/dashmate/src/commands/config/set.js

```diff
@@ -20,7 +21,33 @@ class ConfigSetCommand extends ConfigBaseCommand {
       optionValue = null;
     }
 
-    config.set(optionPath, optionValue);
+    // check for existence
+    config.get(optionPath);
+
+    const path = optionPath.split('.');
+
+    let schema = configJsonSchema;
+
+    for (const e of path) {
+      if (schema.$ref) {
+        const [, , definitionName] = schema.$ref.split('/');
+        schema = configJsonSchema.definitions[definitionName].properties;
+      } else {
+        schema = schema.properties[e];
+      }
+    }
```

Comment:
There is a package that resolves $refs. Check JS DPP

Reply:
I checked and found @apidevtools/json-schema-ref-parser package is used in JS-DPP. But I can't see how it can be useful here. According to documentation, it helps resolving external references that are listed in separate file, urls or other sources. But in our config schema we only use our own definitions that are always listed in our configJsonSchema.js.

When I parse my config schema with that package, it returns me almost identical structure compared to what I have now, it just validated and turnt into classed JS object

Reply:
This code heavily depends on the current schema structure. If I change it to reference to objects outside dependencies or to objects in meta schema the code will break. It's a bad practice and you should avoid of such fragile dependencies in your code. That's why I recommend you to use a library which will support any config structure and will be reliable in the future during the app evolution.